test/functional/shared.js

```diff
@@ -192,12 +192,33 @@ class EventCollector {
   }
 }
 
+function withMonitoredClient(commands, callback) {
+  if (!callback.hasOwnProperty('prototype')) {
+    throw new Error('withMonitoredClient callback can not be arrow function');
+  }
+  return function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ monitorCommands: true });
+    const events = [];
+    client.on('commandStarted', filterForCommands(commands, events));
+    client.connect((err, client) => {
+      expect(err).to.not.exist;
+      const d = function() {
+        const args = Object.values(arguments);
```

Comment:
I think this is different than what made it into [3.6](https://github.com/mongodb/node-mongodb-native/blob/3.6/test/functional/shared.js#L206-L208)

Reply:
Good catch